engine/src/main/java/org/terasology/world/block/IBlockRegion.java

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.world.block;
+
+import org.joml.AABBf;
+import org.joml.LineSegmentf;
+import org.joml.Matrix4fc;
+import org.joml.Planef;
+import org.joml.Rayf;
+import org.joml.Spheref;
+import org.joml.Vector2f;
+import org.joml.Vector3f;
+import org.joml.Vector3fc;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.terasology.entitySystem.entity.EntityRef;
+
+import java.util.Iterator;
+import java.util.Optional;
+
+public interface IBlockRegion extends Iterable<Vector3ic> {
```

Comment:
Just use BlockRegionc. Keep it similar to joml.

Reply:
Yeah, this interface is not meant to be actually in the code - it was just the easiest way to give us something to comment on specific lines :wink: 

`BlockRegionc` would also just contain a subset of these methods (accessing data and performing checks, I guess).

Reply:
BlocRegionc is a read-only interface.